library/src/network.rs

```diff
@@ -171,6 +257,32 @@ pub fn send_patch_check_request(
     return Ok(response);
 }
 
+pub fn report_successful_patch_install(
+    config: &UpdateConfig,
+    state: &UpdaterState,
+    patch_number: usize,
+) -> anyhow::Result<()> {
+    let client_id = state
+        .client_id
+        .clone()
+        .unwrap_or("".to_string())
+        .to_string();
+
+    let event = PatchInstallEvent::new(
+        config.app_id.clone(),
+        current_arch().to_string(),
+        client_id.to_string(),
+        patch_number,
+        current_platform().to_string(),
+        config.release_version.clone(),
+    );
+    let request = CreatePatchInstallEventRequest { event };
+
+    let patch_install_success_fn = config.network_hooks.patch_install_success_fn;
+    let url = &patches_events_url(&config.base_url);
+    patch_install_success_fn(url, request)
```

Comment:
```suggestion
    let url = &patches_events_url(&config.base_url);
    config.network_hooks.patch_install_success_fn(url, request)
```
nit but would this work?

Reply:
It would need to be `(config.network_hooks.patch_install_success_fn)(url, request)`, but yeah, that would work. I did it this way for consistency with the other functions in that file that use network hooks.